File: Kattis/winningwagers.hs
{-# LANGUAGE LambdaCase #-}

import           Data.Function ((&))
import           Data.Functor  ((<&>))

main :: IO ()
main = do
    [_,xs,l] <- getContents <&> lines
    words xs
        & map (\case "COIN" -> 2; "DIE" -> 6; "CARDS" -> 52)
        & product
        & pred
        & (read l*)
        & print

File: Kattis/winningwagers.kt
fun main() {
    readln()
    val chance: Long = readln().split(' ').map { when(it) {
        "COIN" -> 2L
        "DIE" -> 6
        "CARDS" -> 52
        else -> 0
    }}.reduce { a, b -> a*b }
    val wager = readln().toLong()
    println(wager * (chance-1))
}
